spark/src/main/scala/org/apache/spark/sql/delta/catalog/DeltaCatalog.scala

```diff
@@ -629,7 +629,8 @@ class DeltaCatalog extends DelegatingCatalogExtension
     /*
      * WriteBuilder for creating a Delta table.
      */
-    private class DeltaV1WriteBuilder extends WriteBuilder {
+    private class DeltaV1WriteBuilder extends WriteBuilder with SupportsTruncate {
```

Comment:
@cloud-fan this is needed because of https://github.com/apache/spark/commit/5054b41cffc55054918fbabe8981aa9a31f9e2cc. While I do think that the change is correct, I also think we might want to treat a StagingTable slightly different, since it only becomes an actual table once it is committed (truncation just means you don't commit it).

Reply:
Looking at the DS v2 doc
```
  /**
   * Configures a write to replace all existing data with data committed in the write.
   *
   * @return this write builder for method chaining
   */
  WriteBuilder truncate();
```

I think `StagingTable` should implement `SupportsTruncate` to indicate whether it's an append or overwrite. Delta uses table existence to decide append or overwrite, which is also OK but not all the v2 sources can follow it.